client/allocdir/alloc_dir_test.go

```diff
@@ -472,3 +472,18 @@ func TestPathFuncs(t *testing.T) {
 		t.Errorf("%q is not empty. empty=%v error=%v", dir, empty, err)
 	}
 }
+
+func TestAllocDir_DetectContentType(t *testing.T) {
+	require := require.New(t)
+	imgPath := "input/image.png"
+	fileInfo, err := os.Stat(imgPath)
+	require.Nil(err)
+	res := detectContentType(fileInfo, imgPath)
+	require.Equal("image/png", res)
```

Comment:
more nit picking - seems like a good candidate for table testing - I would test go source files and maybe some `application/octet-stream` file as well.